cpp/src/sampling/nbr_sampling_impl.cuh

```diff
@@ -368,103 +375,104 @@ uniform_nbr_sample_impl(
-                             d_in,
-                             d_ranks,
-                             gpu_t{});
+                             d_new_in,
+                             d_new_rank,
+                             std::move(d_rnd_indices),
+                             static_cast<edge_t>(k_level),
+                             global_degree_offsets,
+                             global_adjacency_list_offsets);
+        d_out_src   = std::move(temp_d_out_src);
+        d_out_dst   = std::move(temp_d_out_dst);
+        d_out_ranks = std::move(temp_d_out_ranks);
+        d_indices   = std::move(temp_d_indices);
+      } else {
+        auto&& [temp_d_out_src, temp_d_out_dst, temp_d_out_ranks, temp_d_indices] =
+          gather_one_hop_edgelist(
+            handle, graph_view, d_new_in, d_new_rank, global_adjacency_list_offsets);
+        d_out_src   = std::move(temp_d_out_src);
+        d_out_dst   = std::move(temp_d_out_dst);
+        d_out_ranks = std::move(temp_d_out_ranks);
+        d_indices   = std::move(temp_d_indices);
       }
 
+      // resize accumulators:
+      //
+      auto old_sz = d_acc_dst.size();
+      auto add_sz = d_out_dst.size();
+      auto new_sz = old_sz + add_sz;
+
+      d_acc_src.resize(new_sz, handle.get_stream());
+      d_acc_dst.resize(new_sz, handle.get_stream());
+      d_acc_ranks.resize(new_sz, handle.get_stream());
+      d_acc_indices.resize(new_sz, handle.get_stream());
+
+      // zip quad; must be done after resizing,
+      // because they grow from one iteration to another,
+      // so iterators could be invalidated:
+      //
+      auto acc_zip_it =
+        thrust::make_zip_iterator(thrust::make_tuple(d_acc_src.begin() + old_sz,
+                                                     d_acc_dst.begin() + old_sz,
+                                                     d_acc_ranks.begin() + old_sz,
+                                                     d_acc_indices.begin() + old_sz));
+
+      // union step:
+      //
+      auto out_zip_it = thrust::make_zip_iterator(thrust::make_tuple(
+        d_out_src.begin(), d_out_dst.begin(), d_out_ranks.begin(), d_indices.begin()));
+
+      thrust::copy_n(handle.get_thrust_policy(), out_zip_it, add_sz, acc_zip_it);
+
+      // shuffle step: update input for self_rank
+      // zipping is necessary to preserve rank info during shuffle!
+      //
+      auto next_in_zip_begin =
+        thrust::make_zip_iterator(thrust::make_tuple(d_out_dst.begin(), d_out_ranks.begin()));
+      auto next_in_zip_end =
+        thrust::make_zip_iterator(thrust::make_tuple(d_out_dst.end(), d_out_ranks.end()));
+
+      update_input_by_rank(handle,
+                           graph_view,
+                           next_in_zip_begin,
+                           next_in_zip_end,
+                           static_cast<size_t>(self_rank),
+                           d_in,
+                           d_ranks,
+                           gpu_t{});
+      //}
```

Comment:
Better delete?

Reply:
I will clean up some of the commented out bits in the code.  Part of my work to get things running.